### ISSUE TITLE
7z unpack strategy fails for snap7

### ISSUE THREAD

**Issue by SMillerDev** [Homebrew/homebrew-core#32418]
- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?

To help us debug your issue please explain:
- What you were trying to do (and why)
Install the snap7 formula. Since there's an issue for it open here: https://discourse.brew.sh/t/brew-install-snap7-404-not-found/3052

- What happened (include command output)
When trying to install snap7, the 7z unpackstrategy is somehow not unpacking anything. This causes the rest of the formula to fail.

- What you expected to happen
I expected the 7z unpack strategy to unpack the 7z file.

- Step-by-step reproduction instructions (by running `brew install` commands)
brew install -s snap7

**Gist log**
https://gist.github.com/a37dc38f77978c92eb840df013aea742

**Comment by commitay** [comment]
https://github.com/Homebrew/brew/issues/4980

**Comment by fxcoudert** [comment]
The archive appears valid, it can be unpacked by `7zr` from the command-line. Why `brew` doesn't do it, I do not know.

**Comment by fxcoudert** [comment]
```
==> Verifying e9d50a455993c0b483370b98100fa1ca27ae3443b48d0579af94f39507e74beb--snap7-full-1.4.2.7z checksum
cp -p /Users/fx/Library/Caches/Homebrew/downloads/e9d50a455993c0b483370b98100fa1ca27ae3443b48d0579af94f39507e74beb--snap7-full-1.4.2.7z /private/tmp/snap7-20180925-87509-ro5xcr/snap7-full-1.4.2.7z
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/fileutils.rb:128:in `chdir'
Errno::ENOENT: No such file or directory @ dir_chdir - snap7-full-full
```

**Comment by fxcoudert** [comment]
poke @reitermarkus

**Comment by madeirajp** [comment]
Only to give more information if I try to download via link _https://homebrew.bintray.com/bottles/snap7-full_1.high_sierra.bottle.tar.gz_ I get the message **The requested path was not found.**

**Comment by fxcoudert** [comment]
@madeirajp yes, the bottles are missing (for some reason?) which is what triggers a source build. The problem reported here is that the source does not actually unpack correctly. And we cannot fix the bottles unless we fix that :)

**Comment by fxcoudert** [comment]
We now have:

```
==> Downloading https://downloads.sourceforge.net/project/snap7/1.4.2/snap7-full-1.4.2.7z
Already downloaded: /Users/brew/Library/Caches/Homebrew/downloads/e9d50a455993c0b483370b98100fa1ca27ae3443b48d0579af94f39507e74beb--snap7-full-1.4.2.7z
==> Verifying e9d50a455993c0b483370b98100fa1ca27ae3443b48d0579af94f39507e74beb--snap7-full-1.4.2.7z checksum
env PATH=/usr/local/opt/p7zip/bin:/usr/local/Homebrew/Library/Homebrew/shims/mac/super:/usr/bin:/bin:/usr/sbin:/sbin 7zr x -y -bd -bso0 /Users/brew/Library/Caches/Homebrew/downloads/e9d50a455993c0b483370b98100fa1ca27ae3443b48d0579af94f39507e74beb--snap7-full-1.4.2.7z -o/private/tmp/d20181001-65997-136zzcg
cp -pR /private/tmp/d20181001-65997-136zzcg/snap7-full-1.4.2/. /private/tmp/snap7-20181001-65997-cm0yig/snap7-full-1.4.2
Error: An exception occured within a child process:
  Errno::ENOENT: No such file or directory @ dir_chdir - snap7-full-1.4.2
```

See full logs here: https://jenkins.brew.sh/job/Homebrew%20Mojave%20Testing/759/

**Comment by reitermarkus** [comment]
That's because the filename is wrong.

**Comment by reitermarkus** [comment]
Or rather the `cd` is unnecessary.
